docs/compatibility.md

```diff
@@ -0,0 +1,8 @@
+# Compatibility
```

Comment:
Does it make sense to put this into the README.md to have it visible all the time?

Reply:
is this how it is done in other projects?
Intuitively a permanent and dedicated file under /docs seems cleaner.
But  I have no strong opinion.

Reply:
See https://github.com/kubernetes/client-go/#compatibility-matrix for example